### PR TITLE
i8: add int8 SIMD package (saturating arith, int32 reductions, widening)

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Adds a new `i8` package with seven high-impact int8 SIMD primitives, mirroring the `i16`/`i32` package structure (pure-Go reference, AVX2 amd64 kernels, NEON arm64 kernels, parity/bit-exact/zero-alloc tests, differential fuzzing).

int8's narrow `-128..127` range makes element-wise arithmetic overflow almost immediately, so this package does not mirror the wrapping arithmetic of `i16`/`i32`. It ships the operations that are genuinely high-impact and well-defined at 8-bit width:

- `AddSaturate`, `SubSaturate`: signed saturating add/sub (`VPADDSB`/`VPSUBSB`, `SQADD`/`SQSUB`), clamping to `[-128,127]` rather than wrapping.
- `ToInt16`, `ToInt32`: sign-extending widen (`VPMOVSXB{W,D}`, `SXTL`/`SXTL2`).
- `Sum`, `DotProduct`: int32-accumulated reductions. `DotProduct` is the quantized matmul inner loop: AVX2 widens with `VPMOVSXBW` and reduces with `VPMADDWD`; ARM64 uses an `SDOT` (FEAT_DotProd) fast path with a `SMULL`/`SADALP` base-NEON fallback for cores without it.
- `MinMax`: signed per-slice min/max (`VPMINSB`/`VPMAXSB`, `SMIN`/`SMAX` + `SMINV`/`SMAXV`).

The int32-accumulated reductions wrap in two's complement exactly like their references; int32 wrapping addition is associative, so the lane-parallel SIMD totals are bit-identical to the scalar reference regardless of summation order, and the int8 products never overflow their lane (`|int8 * int8| <= 16384`).

### Supporting changes

- `cpu`: detect `FEAT_DotProd` (`ARM64.DOTPROD` / `HasDOTPROD()`), wired from `x/sys/cpu.HasASIMDDP` on Linux and unconditionally on Apple Silicon. Adds a `dotprod` `SIMD_DISABLE` token, folded into `clearNEON`.
- `asmcheck`: `arm64asm` cannot decode `SDOT`/`UDOT`, so defer those (alongside the existing FP16 `.8H` case) to the aarch64 objdump cross-check.
- `doc.go` / `README.md`: package listing, op tables, per-arch tiers, and a documented follow-up list.

amd64 gates on AVX2 (pure Go below).

## Test Plan

- [x] amd64: full suite green with AVX2 on and with `SIMD_DISABLE=avx2`/`avx`
- [x] arm64 (Cortex-A76, asimddp): full suite green across the SDOT, base-NEON (`SIMD_DISABLE=dotprod`), and pure-Go (`SIMD_DISABLE=neon`) paths
- [x] Parity sweeps across block/tail boundaries, a 140k-element int32 wraparound case, zero-alloc assertions, and four differential fuzzers (no failures)
- [x] `golangci-lint run ./...` clean (whole tree, `new: false`)
- [x] `go vet ./...` clean; asmcheck passes with `SIMD_REQUIRE_OBJDUMP=1` (SDOT verified via objdump)

## Out of scope (documented follow-ups)

- `float32 <-> int8` affine `Quantize`/`Dequantize` (scale + zero-point)
- AVX-512 VNNI (`VPDPBUSD`) `DotProduct` fast path
- 8-bit channel `Interleave2`/`Deinterleave2`

## Summary by CodeRabbit

* **New Features**
  * Added i8 package: SIMD‑accelerated int8 primitives — saturating add/sub, sign‑extending conversions, Sum, DotProduct, and MinMax with zero‑allocation and length‑clamping.

* **Documentation**
  * Updated docs and architecture table to describe i8 support, bit‑exactness, and runtime use of AVX2 on amd64 and an SDOT fast path on capable ARM64 cores.

* **Tests**
  * Added benchmarks, examples, fuzz targets, and unit tests validating correctness, zero‑allocation guarantees, and edge cases.

### CODE REVIEW

File: i8/i8_arm64.s [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    CSEL GT, R7, R5, R5           // clamp high
+    MOVD $-128, R7
+    CMP  R7, R5
+    CSEL LT, R7, R5, R5           // clamp low
+    MOVB R5, (R0)
+    ADD  $1, R1
+    ADD  $1, R2
+    ADD  $1, R0
+    SUB  $1, R3
+    CBNZ R3, addsat_scalar
```
[the remark this answers is not in the document]
Comment (tphakala) [reply i8/i8_arm64.s:58]: Done in 67141d5 - hoisted the 127/-128 clamp constants into R8/R9 before the addSatNEON scalar remainder loop; the loop body only touches R5/R6, so they stay live.

File: i8/i8_arm64.s [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    CSEL GT, R7, R5, R5
+    MOVD $-128, R7
+    CMP  R7, R5
+    CSEL LT, R7, R5, R5
+    MOVB R5, (R0)
+    ADD  $1, R1
+    ADD  $1, R2
+    ADD  $1, R0
+    SUB  $1, R3
+    CBNZ R3, subsat_scalar
```
[the remark this answers is not in the document]
Comment (tphakala) [reply i8/i8_arm64.s:100]: Done in 67141d5 - same hoist for the subSatNEON scalar remainder loop (R8/R9).

File: docs/superpowers/specs/2026-06-13-i8-primitives-design.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+approximately equal, and is the property the parity tests assert.
+
+Intermediate products never overflow their SIMD lane: `|int8 * int8| <= 16384`,
+so the int16 products from `SMULL`/`VPMOVSXBW`+multiply are exact, and a
+`VPMADDWD`/`SADALP` pair-sum (`<= 32768`) is exact in int32. Only the final
+running total can wrap, and it wraps identically to the reference.
+
+## File layout (mirrors i32)
+
+```
```
[the remark this answers is not in the document]
Comment (tphakala) [reply docs/superpowers/specs/2026-06-13-i8-primitives-design.md]: Done in 67141d5 - tagged the file-layout fenced block as `text`.

**Comment by tphakala** [comment]
@coderabbitai addressed the three nitpicks in 67141d5: added `_ = ARM64.DOTPROD` to TestFeatures; extended TestTrailingCapacityUntouched to SubSaturate/ToInt16/ToInt32; and switched the dotAVX2/minMaxAVX2 scalar tails to legacy GPRs (DX, and DI/SI/CX) instead of R8-R11 per the repo's REX-prefix convention.

**Comment by tphakala** [comment]
@coderabbitai review

File: i8/i8_test.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+	if dst16[3] != 4242 || dst16[4] != 4242 {
+		t.Errorf("ToInt16 clobbered trailing capacity: %v", dst16)
+	}
+
+	dst32 := []int32{-9, -9, -9, 424242, 424242}
+	ToInt32(dst32, a)
+	if dst32[3] != 424242 || dst32[4] != 424242 {
+		t.Errorf("ToInt32 clobbered trailing capacity: %v", dst32)
+	}
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply i8/i8_test.go:272]: Done in 8ead640 - TestTrailingCapacityUntouched now loops over sizes {3, 35}, so n=35 exercises the AVX2/NEON kernels (one full block + scalar tail) for all four write APIs, not just the Go fallback.
